src/daemon/rpmostreed-transaction-types.cxx

```diff
@@ -1274,25 +1274,55 @@ deploy_transaction_execute (RpmostreedTransaction *transaction,
           return glnx_throw (error, "No overrides for package '%s'", name_or_nevra);
         }
 
-      changed = TRUE;
-    }
-
-  if (self->override_replace_local_pkgs)
-    {
-      g_autoptr(GPtrArray) pkgs = NULL;
-      if (!import_many_local_rpms (repo, self->override_replace_local_pkgs, &pkgs,
-                                   cancellable, error))
-        return FALSE;
-
-      if (pkgs->len > 0)
+      if (self->override_replace_local_pkgs)
         {
-          g_ptr_array_add (pkgs, NULL);
-          if (!rpmostree_origin_add_overrides (origin, (char**)pkgs->pdata,
-                                               RPMOSTREE_ORIGIN_OVERRIDE_REPLACE_LOCAL,
-                                               error))
+          g_autoptr(GPtrArray) pkgs = NULL;
+          if (!import_many_local_rpms (repo, self->override_replace_local_pkgs, &pkgs,
+                                       cancellable, error))
             return FALSE;
-          changed = TRUE;
-        }
+
+          for (char **it = (char**)pkgs->pdata; it && *it; it++)
+            {
+              const char *name_or_nevra = *it;
+              auto name = static_cast<const char *>(g_hash_table_lookup (nevra_to_name, name_or_nevra));
+              auto nevra = static_cast<const char *>(g_hash_table_lookup (name_to_nevra, name_or_nevra));
+
+              if (name == NULL && nevra == NULL)
+                {
+                  /* it might be an inactive override; just try it both ways */
+                  name = name_or_nevra;
+                  nevra = name_or_nevra;
+                }
+              else if (name == NULL)
+                name = name_or_nevra;
+              else if (nevra == NULL)
+                nevra = name_or_nevra;
+              else
+                {
+                  /* completely brush over the ridiculous corner-case of a
+                     pkgname that's also a nevra for another package */
+                  g_assert_not_reached ();
+                }
+
+          if (rpmostree_origin_remove_override (origin, name,
+                                                RPMOSTREE_ORIGIN_OVERRIDE_REMOVE))
+            continue; /* override found; move on to the next one */
+
+          if (rpmostree_origin_remove_override (origin, nevra,
+                                                RPMOSTREE_ORIGIN_OVERRIDE_REPLACE_LOCAL))
+            continue; /* override found; move on to the next one */
+
```

Comment:
In this case, all we want is to convert the package name to a NEVRA using the `name_to_nevra` table. This will give us the NEVRA of any existing replacement overrides which has a matching package name. Then we'd just need to call `rpmostree_origin_remove_override (..., ..._REPLACE_LOCAL)` if we found one. Does that make sense?

Reply:
That means we only want to replace other `override replace` but we don't want to replace an `override remove` ?

Reply:
Right, that's what I was thinking since they're very different operations, so it's less clear we want that.

/me thinks

Hmm... OK I think this can make sense to do eventually, though if you don't mind, I'd prefer leaving that as a separate RFE. E.g. we'd have to make sure we handle `rpm-ostree override replace foobar-1.2-3.rpm --remove foobar` correctly (i.e. error out).

Reply:
Great, I made the changes and it compiles. But when executing twice `sudo rpm-ostree override replace foobar` the error `error: Bus owner changed, aborting.` appears again. The good part is that means the changes work but also that what we are asking does something forbidden in the way, I think.